specification/elastic/resource-manager/Microsoft.Elastic/preview/2023-06-15-preview/elastic.json

```diff
@@ -0,0 +1,2581 @@
+          "type": "boolean",
+          "description": "IncludeByDefault for the elastic filter"
+        },
+        "rules": {
+          "type": "array",
+          "description": "Rules in the elastic filter",
+          "items": {
+            "$ref": "#/definitions/ElasticTrafficFilterRule"
+          }
+        }
+      }
+    },
+    "ElasticTrafficFilterRule": {
+      "description": "Elastic traffic filter rule object",
+      "type": "object",
+      "properties": {
+        "source": {
+          "type": "string",
+          "description": "IP of the elastic filter rule"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the elastic filter rule"
+        },
+        "azureEndpointGuid": {
+          "type": "string",
+          "description": "Guid of Private Endpoint in the elastic filter rule"
+        },
+        "azureEndpointName": {
+          "type": "string",
+          "description": "Name of the Private Endpoint in the elastic filter rule"
+        },
+        "id": {
+          "type": "string",
+          "description": "Id of the elastic filter rule"
+        }
+      }
+    },
+    "ElasticMonitorUpgrade": {
+      "description": "Upgrade elastic monitor version",
+      "type": "object",
+      "properties": {
+        "version": {
+          "type": "string",
+          "description": "Version to which the elastic monitor should be upgraded to"
+        }
+      }
+    },
+    "UserEmailId": {
+      "description": "Email Id of the User Organization, of which the API Key must be returned",
+      "type": "object",
+      "properties": {
+        "emailId": {
+          "type": "string",
+          "description": "The User email Id"
+        }
+      }
+    },
+    "UserApiKeyResponse": {
+      "description": "The User Api Key created for the Organization associated with the User Email Id that was passed in the request",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/UserApiKeyResponseProperties"
+        }
+      }
+    },
+    "UserApiKeyResponseProperties": {
+      "type": "object",
+      "properties": {
+        "apiKey": {
+          "description": "The User Api Key Generated based on GenerateApiKey flag. This is applicable for non-Portal clients only.",
+          "type": "string",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "ElasticOrganizationToAzureSubscriptionMappingResponse": {
+      "description": "The Azure Subscription ID to which the Organization of the logged in user belongs and gets billed into.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ElasticOrganizationToAzureSubscriptionMappingResponseProperties"
+        }
+      }
+    },
+    "ElasticOrganizationToAzureSubscriptionMappingResponseProperties": {
+      "type": "object",
+      "properties": {
+        "billedAzureSubscriptionId": {
```

Comment:
>billedAzureSubscriptionId [](http://example.com/codeflow?start=9&length=25)

consider add "arm-id" and "x-ms-arm-id-details" for it: [https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-arm-id-details](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-arm-id-details)

Reply:
"x-ms-arm-id-details" is a structure not a "boolean". You can also choose to not set it if the related resource definition not defined by swagger.
[http://azure.github.io/autorest/extensions/#schema](http://azure.github.io/autorest/extensions/#schema)